src/python/pants/core/goals/fmt.py

```diff
@@ -104,8 +106,14 @@ def cacheable(self) -> bool:
 
 
 @union
+@frozen_after_init
+@dataclass(unsafe_hash=True)
 class FmtRequest(StyleRequest):
-    pass
+    prior_formatter_result: Snapshot | None
```

Comment:
Also not a typo. Keeping this as `| None` makes sure `mypy` doesn't complain at the callsites. Will be updating in future PR

Reply:
Please add a comment inline to explain both of these.